### PR TITLE
Directory.Delete(string,bool) Reparse point behavior und fix for recursive=false

### PR DESCRIPTION
**Pull request by SchreinerK**
added: Reparse point behavior for Directory.Delete(string,bool)
added: fix for recursive=false

### CODE REVIEW

**Comment by peteraritchie** [comment]
This PR causes sereral tests to break.  If I find the time, I'll try to fix them; but I might not be able to get to it quickly.

**Comment by zjrunner** [comment]
We were hoping to use this library but this bug is a blocker.  Looks like this PR is 4+ months in -- any chance it will get merged soon so we can avoid building privates?

**Comment by peteraritchie** [comment]
I'll try to find some time to look more into this; but I was hoping the original author would update the PR to fix the broken tests.  4 months ago that might have been faster than me getting up to speed on the PR and updating it to not break tests; but it may be wash at this point.  I'll see if I can find some time to look more closely into this.

File: Pri.LongPath/Directory.cs [hunk 1]
```diff
@@ -47,6 +47,32 @@ public static string GetCurrentDirectory()
...
+			   such as a symbolic link or a mount point. 
+			   (1) If the reparse point is a directory, such as a mount point, it is unmounted and the mount point is deleted. 
+			   This method does not recurse through the reparse point. 
+			   (2) If the reparse point is a symbolic link to a file, the reparse point is deleted and not the target of 
+			   the symbolic link.
+			*/
+
+			try 
+			{
+				var isDirectoryReparsePoint=(Common.GetAttributes(path) & (System.IO.FileAttributes.Directory | System.IO.FileAttributes.ReparsePoint)) != 0;
```
Comment (carl-tanner) [inline Pri.LongPath/Directory.cs]: This should be
```
var reparseFlags = (System.IO.FileAttributes.Directory | System.IO.FileAttributes.ReparsePoint);
var isDirectoryReparsePoint=(Common.GetAttributes(path) & reparseFlags) == reparseFlags;
```
Code in the commit returns true for all directories.

**Comment by carl-tanner** [comment]
There was a bug in the code (which the units rightfully caught).  I put a comment in the first commit from in the pull request that fixes the problem.  All units pass after that change is made.

**Comment by rstarkov** [comment]
Would it help to submit carl-tanner's tweak as a separate PR?

**Pull request merged**